acquia_lift.batch.inc

```diff
@@ -13,16 +13,24 @@
  * @return bool
  *   Returns TRUE if batch syncing completed successfully, FALSE otherwise.
  */
-function acquia_lift_batch_sync_tests_for_agent($agent) {
+function acquia_lift_batch_sync_tests_for_agent($agent, $status_change = NULL) {
   try {
     $operations = acquia_lift_get_sync_operations_for_agent($agent);
-    acquia_lift_do_batch_sync($operations);
-    $error = _acquia_lift_batch_get_error();
-    return !$error;
+    // We don't get to pass additional info to the callback, such as what we'd
+    // like it to do upon success, so we define a different callback depending
+    // on what has been passed as the $status_change parameter.
+    $finish_callback = 'acquia_lift_batch_finish';
+    if ($status_change == PERSONALIZE_STATUS_RUNNING) {
+      $finish_callback = 'acquia_lift_batch_finish_and_start_campaign';
+    }
+    elseif ($status_change == PERSONALIZE_STATUS_SCHEDULED) {
+      $finish_callback = 'acquia_lift_batch_finish_and_schedule_campaign';
+    }
+    $_SESSION['acquia_lift_agent_sync'] = $agent->machine_name;
+    acquia_lift_do_batch_sync($operations, $finish_callback);
   }
   catch (AcquiaLiftException $e) {
     drupal_set_message($e->getMessage(), 'error');
```

Comment:
Since we aren't showing anything else along with this anymore, could we prepend something to indicate that it is a Lift error that prevents synchronization?  Just seeing "No goals have been set up for agent {machine_name}" doesn't really provide enough context for the user I don't think.